### ISSUE TITLE
Extend reported slave state to reflect failures and liveness

### ISSUE THREAD

**Issue by timbod7** [helix-collective/hx-deploy-tool#5]
Currently remote-proxy slaves push their state to S3 on successful completion of a change.  Failures are not reported, requiring ssh'ing into the slave machine for diagnostics.

We will extend the reported slave state to include failures. Instead of uploading a value of type `State` we will upload:

```
struct AtTime<T> {
    Instant timestamp;
    T state;
};

union LastUpdate {
    State Success;
    String Error;     
};

struct SlaveState {
    AtTime<LastUpdate> lastUpdate;
};
```

**Comment by ashvds** [comment]
@timbod7 @jsuresh , believe this is fixed by #69

**Comment by timbod7** [comment]
Yep. resolved with https://github.com/helix-collective/hx-deploy-tool/pull/69
